_posts/2017-06-26-http2.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: post
+title: HTTP2 on MDN
+author: Stephanie Hobson
+excerpt_separator: <!--more-->
+---
+
+We enabled HTTP/2 on MDN's CDN.
+
+We didn't do anything to optimize for HTTP/2, we just enabled it.
+
+We're seeing performance improvements.
+
+<!--more-->
+
+Why is this worth a blog post?
+===
+
+While doing research to see if turning it on without doing any optimizations was
+a good idea I read things like:
+
+> "It also means that all of those HTTP1 performance techniques are harmful.
+> They will make a HTTP2 website slower, not faster - don't use them." -
+> [HTTP2 for front-end web developers](https://mattwilcox.net/web-development/http2-for-front-end-web-developers)
+
+And:
+
+> "However, many of the things you think of as being best practices can be
+> detrimental to performance under HTTP/2." - [Getting Ready For HTTP2: A Guide
+> For Web Designers And Developers](https://www.smashingmagazine.com/2016/02/getting-ready-for-http2/)
+
+Which suggest that enabling HTTP/2 on a site optimized for HTTP/1.1 could result
+in a slower site.
+
+A better way to interpret those quotes is:
+
+> If you optimize for HTTP/1.1 and turn on HTTP/2 your site will not be as fast
+> as it could be - but it might still be faster!
+
+On MDN we concatenate a lot of our files but we don't concatenate all of them.
+For example, our article pages we have 9 different files coming from our CDN. I
+thought we could benefit from a bit of HTTP/2's multiplexing and header
+compression. And we did. You can see the DNS lookup time drop off in this
+waterfall from [Pingdom](https://tools.pingdom.com/):
+
+![Waterfall showing over 0.3s of DNS look up for each request.]({{ site.baseurl }}/public/images/2017-06-26-http2/waterfall_before.png)
+
+![Waterfall showing DNS lookup for only first asset requested.]({{ site.baseurl }}/public/images/2017-06-26-http2/waterfall_after.png)
+
+Some numbers
+===
+
+Overall, our tests don't show a huge improvement in page load speed but there
+are small improvements across the board, and a real impact for users located far
+away from our servers. (Hi Australia and China!)
+
+| Service | Location | Browser | HTTP/1.1 | HTTP/2 | Change |
```

Comment:
I was mentally calculating percents, like 13% speedup for Dallas, and a 44% improvement (!) for Austrailia in GA. Does it make sense to add a percent column?